frontend/src/components/dashboard/create/email/EmailRecipients.tsx

```diff
@@ -63,13 +63,7 @@ const EmailRecipients = ({
   const [preview, setPreview] = useState({} as EmailPreview)
   const { id: campaignId } = useParams()
   const { csvFilename, numRecipients = 0 } = csvInfo
-  const isMounted = useRef(true)
-
-  useEffect(() => {
-    return () => {
-      isMounted.current = false
-    }
-  }, [])
+  const isMounted = useIsMounted()
```

Comment:
As discussed, I think we will need to add this for `SMSRecipients` and `TelegramRecipients` also.